### PR TITLE
Rahulyadav summer

### PR DESCRIPTION
**Pull request by rahulyadav170923**

### CODE REVIEW

File: src/activities/guesscount-summer/Admin.qml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            GCText {
+                anchors.horizontalCenter: parent.horizontalCenter
+                anchors.verticalCenter: parent.verticalCenter
+                text: modelData
+                fontSize: mediumSize
+            }
+            MouseArea {
+                anchors.fill: parent
+                onClicked: {
+                    if(tile.state=="selected"){
```
Comment (petitlapin) [inline src/activities/guesscount-summer/Admin.qml]: put a space before and after "=", "=="
and one after ",", ":" and qml types (not before)
and one before "{" (not after)

File: src/activities/guesscount-summer/Admin.qml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    PropertyChanges { target: tile; color: "green"}
+                },
+                State {
+                    name: "notselected"
+                    PropertyChanges { target: tile; color: "red"}
+                }
+            ]
+        }
+    }
+    Rectangle {
```
Comment (petitlapin) [inline src/activities/guesscount-summer/Admin.qml]: If the window is small, we don't see the text and the last operands.

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        }
+        Timer {
+            id: timer
+            interval: 1500
+            repeat: false
+            onTriggered: {
+                items.solved=true
+                if(items.sublevel<items.data.length)
+                {
+                    Activity.next_sublevel()
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: camel case, should be nextSublevel

File: src/activities/guesscount-summer/dataset.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+// questions according to operators
+
+var dataset=[
+            [
+                [['+'],[  [[1,2],3] , [[3,4],7] ,[[6,2],8] ,[[5,2],7] ,[[9,2],11] ,[[4,2],6],[[1,6],7] ,[[3,2],5] ,[[13,1],14] ,[[5,8],13]  ]],
+                [['-'],[  [[2,1],1] , [[3,1],2] ,[[7,2],5] ,[[6,2],4] ,[[9,2],7] ,[[8,5],3],[[9,5],4],[[18,9],3],[[5,1],4],[[6,3],3],[[5,4],1] ]],
```
Comment (petitlapin) [inline src/activities/guesscount-summer/dataset.js]: [[18,9],9]
for all the dataset, you should add fake numbers (at least one for first level) else it's too easy with only two numbers to guess the result.

File: src/activities/guesscount-summer/guesscount-summer.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+ */
+.pragma library
+.import QtQuick 2.0 as Quick
+.import GCompris 1.0 as GCompris
+.import "dataset.js" as Data
+.import "qrc:/gcompris/src/core/core.js" as Core
+
+
+var url = "qrc:/gcompris/src/activities/guesscount-summer/resource/"
+var operators = [
```
Comment (petitlapin) [inline src/activities/guesscount-summer/guesscount-summer.js]: unused?

File: src/activities/guesscount-summer/guesscount-summer.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+            return false
+        }
+    }
+    return true
+}
+
+
+function equal(levelOperators,array){
+    for(var i in levelOperators){
+        var found=0
```
Comment (petitlapin) [inline src/activities/guesscount-summer/guesscount-summer.js]: you can use a bool there

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+            Rectangle{
+                width: parent.width*0.35;
+                height: parent.height;
+                radius: 20
+                color: "orange"
+                anchors{
+                    right: parent.right
+                    rightMargin: parent.width*0.028
+                }
+                GCText{
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: if you reduce the size screen, the text is bigger than the rectangle

File: src/activities/guesscount-summer/dataset.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                [['*'],[  [[2,1],2] , [[3,1],3] ,[[7,2],14] ,[[6,2],12] ,[[9,2],18] ,[[8,5],40],[[9,5],45],[[1,9],9],[[5,1],5],[[6,3],18],[[5,4],20] ]],
+                [['/'],[  [[2,1],2] , [[3,1],3] ,[[8,2],4] ,[[6,2],3] ,[[9,3],3] ,[[8,4 ],2],[[9,3],3],[[18,9],2]    ]]
+            ],
+
+            [
+                [['+','-'],[  [[1,2,4],1] , [[1,2,4],5] ,[[3,4,5],2] ,[[8,7,9],6] ,[[9,4,6],7] ,[[3,4,5],6] ,[[6,4,5],5]  ]],
```
Comment (petitlapin) [inline src/activities/guesscount-summer/dataset.js]: Is [[1,2,4],1] with '+', '-' doable?
Same for [[3,3,6],9] and [[14,7,3],21] with '*', '/'?

Also, we should be able to use the same operator multiple times.

File: src/activities/guesscount-summer/dataset.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+            [
+                [['+','-','*','/'],[  [[1,7,2,3,2],13] ,[[8,1,5,2,3],6] ,[[1,7,2,4,2],5] ,[[9,7,8,2,4],4]]]
+            ]
+
+        ]
+
+// no of Question at each level
+
+var levelSchema = [4,4,4,4,3,3]
```
Comment (petitlapin) [inline src/activities/guesscount-summer/dataset.js]: I'm not sure if it is useful to limit the number of questions for each level?

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+            property alias operandRow : operandRow
+            property  var data
+            property int result: data[sublevel-1][1]
+            property alias timer: timer
+            property alias dialog: dialog
+            property GCAudio audioEffects: activity.audioEffects
+            property bool solved
+            property bool levelchanged : false
+            property var levelArr
+            property alias load: dialogActivityConfig
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: not used?

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    anchors.verticalCenter: parent.verticalCenter
+                    fontSize: mediumSize
+                    text: qsTr("Guesscount: %1").arg(items.result)
+                }
+            }
+        }
+
+        Column {
+            spacing: 10
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: if you reduce the screen at minimum and have all the operands (let's say that for levle one, you selected admin mode and all the operands), we don't see the last numbers.

**Comment by rahulyadav170923** [comment]
ok Johhny i will correct these errors

**Comment by rahulyadav170923** [comment]
hey Johhny . I have made the changes Pls check it out.
We need the level schema to choose as many sublevels as we need for a
level(they are choosen  randomly from the lot )
and i will form new questions after talking to emmanuel.

Thanks

**Comment by petitlapin** [comment]
Hi,

I quickly tested and I found a little issue: at level 5, if I have: 2 1 4 + - and need to find 5. I put:
2 [] 1 = []
[] + 4 = []
If I put - between 2 and 1, the result is well printed and expected but I don't go to next level.

Also, on admin mode, the flow contentHeight seems to small, I can't see the latest line completely.

But, it's really small bugs, except this all the other issues seem fixed.

I'll take a closer look later.

Johnny

**Comment by petitlapin** [comment]
Hi, I just tested on my phone/tablet. It misses a default mode also (else there is nothing displayed!)

**Comment by rahulyadav170923** [comment]
i have solved the previous issues. i'll solve this default mode problem and
then commit the changes.

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+                        }
+                        Rectangle {
+                            width: parent.width
+                            color: "transparent"
+                            height: parent.height/1.25-labels.height-modeBox.height
+                            anchors {
+                                top: labels.bottom
+                                topMargin: 5
+                            }
+                            ListView {
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: In landscape mode (on phone and tablet), I still don't see entirely the last line.

File: src/activities/guesscount-summer/GuesscountSummer.qml [hunk 13]
```diff
@@ -0,0 +1,374 @@
...
+            height: parent.height
+            OperatorRow {
+                id: operatorRow
+                width: parent.width
+                height: parent.height/10
+                mode: items.mode
+                operators: items.levelArr
+                level: items.currentlevel
+            }
+            OperandRow {
```
Comment (petitlapin) [inline src/activities/guesscount-summer/GuesscountSummer.qml]: at last level, when there are 5 numbers, the last one is not fully displayed on phone/tablet in portrait mode)

**Comment by rahulyadav170923** [comment]
ok . i'll get these errors resolved .Thanks for the review.

**Comment by petitlapin** [comment]
Hi,

I found new bugs:
* if I finish 1 complete level (all 4 sublevels), when I finish the first one in the next level, the operators stay in place. It does not happen if I select next level manually.
* the rectangle is too small for the error "result is not a positive integer". Instead of making it disappears (too quickly), I would wait for a user click (I did the same kind of error in share activity where the text size adapts to the rectangle).

Johnny

**Pull request closed without merging**